### PR TITLE
doc: Added Java collection modeling

### PR DESCRIPTION
**Pull request by JigarJoshi**

### CODE REVIEW

File: docs/datamodels/using-java.md [hunk 1]
```diff
@@ -1 +1,256 @@
...
+The name of the collection is derived from the Java class name. The class name
+name is pluralized to snake_cases as collection name. For example, the
+Java class name `User` is converted to `users` as the collection name. While
+the Java class name `UserDetail` is converted to `user_details` as the
+collection name. User can optionally customize their collection name by
+using `@TigrisCollection` annotation. For example below will create Tigris
+collection named `user_master`.
+
+```java
+@TigrisCollection("user_master")
```
Comment (ovaistariq) [inline docs/datamodels/using-java.md:78]: Don't use the term master.

File: docs/datamodels/using-java.md [hunk 2]
```diff
@@ -1 +1,256 @@
...
+    @TigrisPrimaryKey(1)
+    private int id;
+
+    @TigrisField(description = "Name of the user")
+    private String name;
+
+    @TigrisField(description = "User account balance")
+    private double balance;
+
+    // constructors + accessors + equals + hashcode + toString
```
Comment (ovaistariq) [inline docs/datamodels/using-java.md:35]: Let's all the code here including constructor, accessor and hash code. The idea being that I should be able to copy this piece of code in my Java app

File: docs/datamodels/using-java.md [hunk 3]
```diff
@@ -1 +1,256 @@
...
+  @TigrisField(description = "Name of the product")
+  private String name;
+
+  @TigrisField(description = "Number of products available in the store")
+  private int quantity;
+
+  @TigrisField(description = "Price of the product")
+  private double price;
+
+  // constructors + accessors + equals + hashcode + toString
```
Comment (ovaistariq) [inline docs/datamodels/using-java.md:184]: Put all the needed code here

File: docs/datamodels/using-java.md [hunk 4]
```diff
@@ -1 +1,256 @@
...
+}
+
+
+public class Order implements TigrisCollectionType {
+    @TigrisPrimaryKey(1)
+    private long id;
+    private long userId;
+    private Product[] products;
+
+    // constructors + accessors + equals + hashcode + toString
```
Comment (ovaistariq) [inline docs/datamodels/using-java.md:194]: Add all the needed code here to show what actual class will look like

File: docs/datamodels/using-java.md (hunk 2 above)
Comment (JigarJoshi) [reply docs/datamodels/using-java.md]: updated

**Review by ovaistariq — APPROVED** [review]

**Pull request merged**